src/wallet/rpcnames.cpp

```diff
@@ -628,10 +646,12 @@ name_update ()
       if (!coinsTip.GetName (name, oldData) || oldData.isExpired ())
         throw JSONRPCError (RPC_TRANSACTION_ERROR,
                             "this name can not be updated");
-
+      if (isDefaultVal)
+        {
+          value = oldData.getValue();
+        }
```

Comment:
Tiny nit: Since this is just a single-line body, I would remove the curly braces (similar to the `throw` just above).